### PR TITLE
[AoG] Fix two race conditions in amp-bind/amp-list when backgrounded

### PR DESCRIPTION
**Pull request by choumx**
amp-bind: 
- During amp-bind initialization, force query-able `<amp-state>` to build instead of waiting on the runtime, which can stall when (A) page transitions from background and (B) viewer is hidden. 

amp-list: 
- When using `premutate`, `mutatedAttributesCallback` can be called on un-laid-out `amp-list` elements. This causes a runtime error when trying to render from local state. Instead, defer render until `layoutCallback`.

Also, cleans up some not-useful log statements.

`b/129297489`

### CODE REVIEW

File: extensions/amp-bind/0.1/bind-impl.js [hunk 1]
```diff
@@ -475,39 +475,26 @@ export class Bind {
...
-   */
-  onReady_() {
-    this.viewer_.sendMessage('bindReady', undefined);
-    this.dispatchEventForTesting_(BindEvents.INITIALIZE);
+    }).then(() => {
+      const ampStates = root.querySelectorAll('AMP-STATE');
+      // Force all query-able <amp-state> elements to be built.
+      // A bit weird but amp-state is just metadata, so there's no point
+      // to wait for runtime to schedule builds before sending "bindReady".
+      const whenBuilt = toArray(ampStates).map(el => el.build());
```
Comment (jridgewell) [inline extensions/amp-bind/0.1/bind-impl.js]: This doesn't synchronize build state with resource state, build has to come down from resources. I'm a bit confused why we need to force build, though, or why amp-state can't be a prerenderable element itself.

Comment (choumx) [reply extensions/amp-bind/0.1/bind-impl.js]: Changed this to be an API on amp-state element itself instead as discussed.

File: extensions/amp-bind/0.1/bind-impl.js [hunk 2]
```diff
@@ -475,39 +475,27 @@ export class Bind {
...
-   */
-  onReady_() {
-    this.viewer_.sendMessage('bindReady', undefined);
-    this.dispatchEventForTesting_(BindEvents.INITIALIZE);
+    }).then(() => {
+      const ampStates = root.querySelectorAll('AMP-STATE');
+      // Force all query-able <amp-state> elements to parse local data instead
+      // of waiting for runtime to build them all.
+      const whenParsed = toArray(ampStates).map(el => {
+        return el.getImpl().then(impl => impl.parseAndUpdate());
```
Comment (jridgewell) [inline extensions/amp-bind/0.1/bind-impl.js]: I'm not sure the elements are guaranteed to be upgraded to custom elements yet. You'll need to [wait for upgrade](https://github.com/ampproject/amphtml/blob/bf504ab750f64b9343c228a1756d38d95bbba7c8/src/dom.js#L758-L773) first.

**Pull request merged**